From a370686d19d250e459335b10e4a4248319725bce Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sujeet.baranwal@cavium.com>
Date: Tue, 27 Nov 2018 04:01:10 +0300
Subject: [PATCH 0777/1051] mmc: octeontx2: add check for 8xxx chips

Implement a function which tests for the
underlying chip being 8xxx or not. This
API is useful on implementing features
specific to the octtx2 chips.

Signed-off-by: Sujeet Baranwal <sujeet.baranwal@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/mmc/host/cavium.h b/drivers/mmc/host/cavium.h
index d4b6377c11e6..33b6f1b07975 100644
--- a/drivers/mmc/host/cavium.h
+++ b/drivers/mmc/host/cavium.h
@@ -24,6 +24,9 @@
 #define CAVIUM_MAX_MMC		4
 
 /* Subsystem Device ID */
+#define PCI_SUBSYS_DEVID_88XX   0xA100
+#define PCI_SUBSYS_DEVID_81XX   0xA200
+#define PCI_SUBSYS_DEVID_83XX   0xA300
 #define PCI_SUBSYS_DEVID_96XX	0xB200
 #define PCI_SUBSYS_DEVID_95XX   0xB300
 
@@ -224,6 +227,15 @@ int cvm_mmc_of_slot_probe(struct device *dev, struct cvm_mmc_host *host);
 int cvm_mmc_of_slot_remove(struct cvm_mmc_slot *slot);
 extern const char *cvm_mmc_irq_names[];
 
+static inline bool is_mmc_8xxx(struct cvm_mmc_host *host)
+{
+	struct pci_dev *pdev = host->pdev;
+
+	return (pdev->subsystem_device == PCI_SUBSYS_DEVID_81XX) ||
+		(pdev->subsystem_device == PCI_SUBSYS_DEVID_83XX) ||
+		(pdev->subsystem_device == PCI_SUBSYS_DEVID_88XX);
+}
+
 static inline bool is_mmc_otx2(struct cvm_mmc_host *host)
 {
 	struct pci_dev *pdev = host->pdev;
-- 
2.17.1

